datacatalog/pkg/manager/impl/artifact_manager.go

```diff
@@ -416,6 +422,67 @@ func (m *artifactManager) UpdateArtifact(ctx context.Context, request *datacatal
 	}, nil
 }
 
+func (m *artifactManager) deleteArtifact(ctx context.Context, datasetID *datacatalog.DatasetID, queryHandle artifactQueryHandle) error {
+	ctx = contextutils.WithProjectDomain(ctx, datasetID.Project, datasetID.Domain)
+
+	// artifact must already exist, verify first
+	artifactModel, err := m.findArtifact(ctx, datasetID, queryHandle)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to get artifact while trying to delete [%v], err: %v", queryHandle, err)
+		return err
+	}
+
+	// delete all artifact data from the blob storage
```

Comment:
Note: this isn't ideal that this isn't done in the same transaction as deleting the artifact from DB